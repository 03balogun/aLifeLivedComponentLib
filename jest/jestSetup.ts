/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
