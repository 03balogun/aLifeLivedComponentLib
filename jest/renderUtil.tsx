import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
