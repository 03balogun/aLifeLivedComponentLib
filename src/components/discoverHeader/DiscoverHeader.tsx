import React from 'react';
import { Box } from '../../themes/box/Box';
import { SystemButton } from '../../themes/systemButton/SystemButton';
import { Text } from '../../themes/text/Text';
import { Avatar } from '../avatar/Avatar';
import { Icon } from '../icon/Icon';

type DiscoverHeaderProps = {
  fullName: string;
  avatarUrl: string;
  avatarTestID: string;
  searchTestID: string;
  onSearchIconPress: () => void;
  onAvatarPress: () => void;
};

export const DiscoverHeader: React.FC<DiscoverHeaderProps> = ({
  fullName,
  avatarUrl,
  avatarTestID,
  searchTestID,
  onSearchIconPress,
  onAvatarPress,
}) => {
  return (
    <Box
      flexDirection="row"
      justifyContent="space-between"
      alignContent="center"
      alignItems="center"
    >
      <SystemButton
        testID={avatarTestID}
        flexDirection="row"
        alignItems="center"
        onPress={onAvatarPress}
      >
        <Avatar
          source={{ uri: avatarUrl, priority: 'high' }}
          iconHeight={40}
          iconWidth={40}
        />
      </SystemButton>
      <Box marginLeft="s" width="75%">
        <Text variant="h4" numberOfLines={1}>
          Hi, {fullName}
        </Text>
      </Box>
      <SystemButton testID={searchTestID} onPress={onSearchIconPress}>
        <Icon size={24} icon="search" color="backgroundContrastMid" />
      </SystemButton>
    </Box>
  );
};
