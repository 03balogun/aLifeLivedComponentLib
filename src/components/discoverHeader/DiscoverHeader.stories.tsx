import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

// Component
import { DiscoverHeader } from './DiscoverHeader';
import { Box } from '../../themes/box/Box';

storiesOf('Headers', module)
  .addDecorator(Story => Story())
  .add('Discover header', () => (
    <Box margin="m">
      <DiscoverHeader
        fullName={text('Title', 'John Smith')}
        avatarUrl=""
        onAvatarPress={action('Avatar press')}
        onSearchIconPress={action('Search icon press')}
        avatarTestID="avatar-test-id"
        searchTestID="search-test-id"
      />
    </Box>
  ));
