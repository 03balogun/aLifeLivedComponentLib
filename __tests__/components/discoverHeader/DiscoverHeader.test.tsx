import React from 'react';
import { render, fireEvent } from './../../../jest/renderUtil';

// Component
import { DiscoverHeader } from '../../../src/components/discoverHeader/DiscoverHeader';

const onSearchIconPressMock = jest.fn();
const onAvatarPressMock = jest.fn();

const setup = () =>
  render(
    <DiscoverHeader
      fullName="John Smith"
      avatarUrl=""
      avatarTestID="avatar-test-id"
      searchTestID="search-icon-test-id"
      onSearchIconPress={onSearchIconPressMock}
      onAvatarPress={onAvatarPressMock}
    />
  );

describe('Given discover header is rendered', () => {
  test('Should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should find welcome message', () => {
    const { getByText } = setup();
    expect(getByText('Hi, John Smith')).toBeDefined();
  });

  test('Should find avatar by test id', () => {
    const { getByTestId } = setup();
    expect(getByTestId('avatar-test-id')).toBeDefined();
  });

  test('Should find search icon by test id', () => {
    const { getByTestId } = setup();
    expect(getByTestId('search-icon-test-id')).toBeDefined();
  });

  describe('Given user presses avatar', () => {
    test('Should fire on avatar press', () => {
      const { getByTestId } = setup();
      fireEvent.press(getByTestId('avatar-test-id'));
      expect(onAvatarPressMock).toBeCalledTimes(1);
    });
  });

  describe('Given user presses search icon', () => {
    test('Should fire on search icon press', () => {
      const { getByTestId } = setup();
      fireEvent.press(getByTestId('search-icon-test-id'));
      expect(onSearchIconPressMock).toBeCalledTimes(1);
    });
  });
});
