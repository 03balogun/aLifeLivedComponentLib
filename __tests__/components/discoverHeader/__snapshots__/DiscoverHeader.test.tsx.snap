// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given discover header is rendered Should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
    testID="avatar-test-id"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#67B6A8",
            "borderRadius": 96,
            "borderWidth": 0,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "padding": 16,
            "width": 40,
          },
          Array [
            Object {},
          ],
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "marginLeft": 8,
          "width": "75%",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontFamily": "SFUIText-Regular",
            "fontSize": 20,
          },
        ]
      }
    >
      Hi, 
      John Smith
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="search-icon-test-id"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#CBCBCB"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={24}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={512}
      vbWidth={512}
      width={24}
    >
      <RNSVGGroup
        fill={4291546059}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill={4291546059}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
